src/components/TodoItem/TodoLoader.tsx

```diff
@@ -0,0 +1,25 @@
+import cn from 'classnames';
+import { useTodosContext } from '../../context/TodosContext';
+import { Todo } from '../../types/Todo';
+
+type TodoLoaderProps = {
+  todo: Todo;
+};
+
+export const TodoLoader: React.FunctionComponent<TodoLoaderProps> = ({
+  todo,
+}) => {
+  const { loadingIds } = useTodosContext();
+
+  return (
+    <div
+      data-cy="TodoLoader"
+      className={cn('modal overlay', {
+        'is-active': loadingIds?.includes(todo.id),
```

Comment:
Looking at your types, you should not need `?` in `loadingIds?.includes`

If your code actually crashes without it, this may be a sign of some bug

Probably non-mandatory fix, but it makes sense to re-check